Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM amazon/opendistro-for-elasticsearch:1.12.0
+
+WORKDIR /usr/share/elasticsearch
+
+ENV PATH=/usr/share/elasticsearch/bin:$PATH
+
+RUN elasticsearch-plugin install --batch repository-s3
```

Comment:
I have a doubt about this. What version of the plugin is installed? The latest one or the one linked to that version?

Reply:
the one linked to that version of Elasticsearch. Since Opendistro 1.12.0 has Elasticsearch 7.10.0, the plugins have this last version too
```
GET _cat/plugins
googledemo-odfe-0 opendistro-anomaly-detection    1.12.0.0
googledemo-odfe-0 opendistro-job-scheduler        1.12.0.0
googledemo-odfe-0 opendistro-knn                  1.12.0.0
googledemo-odfe-0 opendistro-reports-scheduler    1.12.0.0
googledemo-odfe-0 opendistro_alerting             1.12.0.2
googledemo-odfe-0 opendistro_index_management     1.12.0.1
googledemo-odfe-0 opendistro_performance_analyzer 1.12.0.0
googledemo-odfe-0 opendistro_security             1.12.0.0
googledemo-odfe-0 opendistro_sql                  1.12.0.0
googledemo-odfe-0 repository-gcs                  7.10.0
googledemo-odfe-0 repository-s3                   7.10.0
```

Reply:
Ok. I'm just concerned about using the right versions of the packages because later versions are not open source/free software. I'm afraid we might end installing versions that we don't really want.

Reply:
I'd rather to install the versions that we know they are open source, to avoid any type of problem. For example, https://artifacts.elastic.co/downloads/elasticsearch-plugins/repository-s3/repository-s3-7.10.0.zip. What do you think?

Reply:
With core plugins like these, we are already ensuring that by installing Open Distro 1.12.0. Which installs Elasticsearch 7.10.0. Which in turn installs 7.10.0 versioned core plugins. Core plugin versions[ always match the Elasticsearch version they are installed in](https://www.elastic.co/guide/en/elasticsearch/plugins/7.10/intro.html). Plugins from other versions are not compatible:
> Core Plugins
> This category identifies plugins that are part of Elasticsearch project. Delivered at the same time as Elasticsearch, their version number always matches the version number of Elasticsearch itself.

They allow offline install for caching reasons. But otherwise, online and offline install are the same. The online (non URL) installs the only compatible version. The offline (with URL) only allows installing the only compatible version.

Reply:
Understood. Let's keep it this way then. Thanks for the clarification.